adaptive/learner/sequence_learner.py

```diff
@@ -15,18 +15,19 @@ class _IgnoreFirstArgument:
     pickable.
     """
 
-    def __init__(self, function):
+    def __init__(self, function, sequence):
         self.function = function
+        self.sequence = sequence
 
-    def __call__(self, index_point, *args, **kwargs):
-        index, point = index_point
+    def __call__(self, index, *args, **kwargs):
+        point = self.sequence[index]
         return self.function(point, *args, **kwargs)
```

Comment:
Maybe make a one-liner
```suggestion
        return self.function(self.sequence[point], *args, **kwargs)
```